blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -117,34 +132,26 @@ public void insertMetrics( Collection<IMetric> metrics ) throws IOException {
 
         for ( Locator locator : locators ) {
 
-            Object type;
+            DataType metricType;
             try {
-
-                type = metadataCache.get( locator, DATA_TYPE_CACHE_KEY );
-
+                metricType = getDataType(locator);
             } catch ( CacheException e ) {
                 LOG.error(String.format("Error looking up locator %s in cache", locator), e);
                 unknowns.add( locator );
                 continue;
             }
 
-
-            DataType metricType = new DataType( (String) type );
-
-            if ( type == null || !DataType.isKnownMetricType( metricType ) ) {
```

Comment:
@usnavi this logic works for Splot ROR use cases but breaks for normal rollup. It is because in the Astyanax world, there are really 2 methods of ```getDatapointsForRange```. One takes a single locator. Another takes a list of locator. The two methods do NOT behave the same wrt to unknown metric data type. 

The getDatapointsForRange() with single Locator is called from RollupHandler.createRepairPoints() when Splot ROR is requested. Here, if data type is unknown or null, we assume it's Numeric.

The getDatapointsForRange() with a list of Locators is called from RollupHandler.getRollupByGranularity() during normall rollup. Here, we don't check for data type being null. We check only RollupType.